core_api_c_binding/src/lib.rs

```diff
@@ -25,23 +32,42 @@ impl Logger for NullLogger {
 
 #[no_mangle]
 pub unsafe extern "C" fn holochain_new(ptr: *mut Dna) -> *mut Holochain {
-    let agent = Agent::from("c_bob".to_string());
-
-    let context = Arc::new(Context::new(
-        agent,
-        Arc::new(Mutex::new(NullLogger {})),
-        Arc::new(Mutex::new(SimplePersister::new())),
-    ));
+    let context = get_context();
 
     assert!(!ptr.is_null());
     let dna = Box::from_raw(ptr);
 
-    match Holochain::new(*dna, context) {
-        Ok(hc) => Box::into_raw(Box::new(hc)),
+    match context {
```

Comment:
```suggestion
context.and_then(|con| Holochain::new(*dna, Arc::new(con))
    .and_then(|hc| Box::into_raw(Box::new(hc)))
).unwrap_or_else(|| std::ptr::null_mut())
```

Reply:
I thought it might be bothersome to always sync up the last two lines (the ones with the `ptr::null_mut()`)